@rfranzke @mbohlool @stts @idealhack Absence of OpenAPI scheme methods for metav1.Duration also causing issues with generating OpenAPI code for CRDs that are using metav1 API.
```
apis/config/v1alpha1/openapi_generated.go:13098:31: v1.Duration literal.OpenAPISchemaType undefined (type v1.Duration has no field or method OpenAPISchemaType)
apis/config/v1alpha1/openapi_generated.go:13099:31: v1.Duration literal.OpenAPISchemaFormat undefined (type v1.Duration has no field or method OpenAPISchemaFormat)
```
To workaround this issue I use forked kubernetes/apimachinery with applied patch from this PR, but it's quite a dirty hack. What should we do to merge this PR?

--- a/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/duration.go
+++ b/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/duration.go
@@ -48,3 +48,13 @@ func (d *Duration) UnmarshalJSON(b []byte) error {
 func (d Duration) MarshalJSON() ([]byte, error) {
 	return json.Marshal(d.Duration.String())
 }
+
+// OpenAPISchemaType is used by the kube-openapi generator when constructing
+// the OpenAPI spec of this type.
+//
+// See: https://github.com/kubernetes/kube-openapi/tree/master/pkg/generators
+func (_ Duration) OpenAPISchemaType() []string { return []string{"string"} }
+
+// OpenAPISchemaFormat is used by the kube-openapi generator when constructing
+// the OpenAPI spec of this type.
+func (_ Duration) OpenAPISchemaFormat() string { return "" }

